src/subgraph-client/subgraph-client.service.spec.ts

```
@@ -28,18 +28,36 @@
+
+    // Mock the decoded data for the ethers AbiCoder
+    const mockDecodedData = [
+      [
+        ['0xuser1', 'tariAddress1'], // tuple(address,string)
+        { toString: () => '1000000000000000000' }, // uint256 amount
+      ],
+    ];
+
+    // Mock for ethers.utils.AbiCoder
+    jest
+      .spyOn(require('ethers').utils.AbiCoder.prototype, 'decode')
```

ESLint

[error] Require statement not part of import statement.

(@typescript-eslint/no-var-requires)